nntrainer/compiler/slice_realizer.cpp

```diff
@@ -10,17 +10,28 @@
  * @bug No known bugs except for NYI items
  */
 
+#include <connection.h>
+#include <iterator>
 #include <layer_node.h>
 #include <slice_realizer.h>
 
 #include <unordered_map>
 
 namespace nntrainer {
 
-SliceRealizer::SliceRealizer(const std::vector<std::string> &start_layers,
-                             const std::vector<std::string> &end_layers) :
-  start_layers(start_layers),
-  end_layers(end_layers.begin(), end_layers.end()) {}
+SliceRealizer::SliceRealizer(const std::vector<Connection> &start_layers,
+                             const std::vector<Connection> &end_layers) {
+  /// discard index information as it is not needed as it is not really needed
+  this->start_layers.reserve(start_layers.size());
+
+  std::transform(start_layers.begin(), start_layers.end(),
+                 std::back_inserter(this->start_layers),
+                 [](const Connection &c) { return c.getName(); });
```

Comment:
let's issue a warning if the user provides a specific connection for a layer as it is not possible to slice a layer for just that connection. (start_layer {"a(0)"}, {"a"}, {"a(1)"} are all treated equally but caller can have different intention).

Reply:
Could you elaborate on the intention of the given example?

My thinking is that it would be slicing A but could not infer more than that.

Reply:
So, a(2) might give the perception to the user that layer A will be sliced internally and only a(2) will remain but we dont support that. As `(2)` is not being considered, add a warning.
Let me know if I misunderstood.

Reply:
I think there is no meaning for the warning. 
a(0) and a(1) will remain as per your understanding. but leaving a(0) and a(1) is a must to make a(2) third input, this imposes there will be a(0) and a(1) and there is no way to slice them away in a sensible manner anyway, unless what you'd want to point out is a(2) becomes a(0) given only a(2) is only slice start.